File: grpc/examples/go/get-orders-and-trades-for-party.go
package main

import (
	"fmt"

	"github.com/vegaprotocol/api/grpc/clients/go/generated/code.vegaprotocol.io/vega/proto/api"
	"github.com/vegaprotocol/api/grpc/examples/go/helpers"

	"golang.org/x/net/context"
	"google.golang.org/grpc"
)

func main() {
	// Helpers include logic to authenticate with a Vega wallet service
	// including token storage/public key operations/signing of commands
	// visit ./helpers/wallet.go for more detail
	pubkey := helpers.GetPubKey()

	nodeURLGrpc := helpers.GetFromEnv("NODE_URL_GRPC")
	conn, err := grpc.Dial(nodeURLGrpc, grpc.WithInsecure())
	if err != nil {
		panic(err)
	}
	defer conn.Close()

	dataClient := api.NewTradingDataServiceClient(conn)

	// __get_orders_for_party:
	// Request a list of orders by party (pubKey)
	ordersByPartyReq := api.OrdersByPartyRequest{PartyId: pubkey}
	ordersByPartyResp, err := dataClient.OrdersByParty(context.Background(), &ordersByPartyReq)
	if err != nil {
		panic(err)
	}
	fmt.Printf("OrdersByParty: %v\n", ordersByPartyResp)
	// :get_orders_for_party__

	// __get_trades_for_party:
	//Request a list of trades by market on a Vega network
	tradesByPartyReq := api.TradesByPartyRequest{PartyId: pubkey}
	tradesByPartyResp, err := dataClient.TradesByParty(context.Background(), &tradesByPartyReq)
	if err != nil {
		panic(err)
	}
	fmt.Printf("TradesByParty: %v\n", tradesByPartyResp)
	// :get_trades_for_party__
}

File: grpc/examples/python/get-orders-and-trades-for-party.py
#!/usr/bin/python3

"""
Script language: Python3

Talks to:
- Vega wallet (REST)
- Vega node (gRPC)

Apps/Libraries:
- REST: requests (https://pypi.org/project/requests/)
"""

# Note: this file uses smart-tags in comments to section parts of the code to
# show them as snippets in our documentation. They are not necessary to be
# included when creating your own custom code.
#
# Example of smart-tags:
#  __something:
# some code here
# :something__

import helpers

# Vega wallet interaction helper, see login.py for detail
from login import token, pubkey

# __import_client:
import vegaapiclient as vac
node_url_grpc = helpers.get_from_env("NODE_URL_GRPC")
data_client = vac.VegaTradingDataClient(node_url_grpc)
# :import_client__

print("Token:", token)

# __get_orders_for_party:
# Request a list of orders by party (pubKey)
orders_by_party_request = vac.api.trading.OrdersByPartyRequest(
    party_id=pubkey
)
orders_response = data_client.OrdersByParty(orders_by_party_request)
print("OrdersByParty:\n{}".format(orders_response))
# :get_orders_for_party__

# __get_trades_for_party:
# Request a list of trades by party (pubKey)
trades_by_party_request = vac.api.trading.TradesByPartyRequest(
    party_id=pubkey
)
trades_response = data_client.TradesByParty(trades_by_party_request)
print("TradesByParty:\n{}".format(trades_response))
# :get_trades_for_party__
